app/services/smart_proxies/puppet_ca_certificate.rb

```diff
@@ -0,0 +1,30 @@
+require "time"
+
+class SmartProxies::PuppetCACertificate
+  attr_reader :name, :state, :fingerprint, :valid_from, :expires_at, :smart_proxy_id
```

Comment:
imho you wont need to pass the proxy id, rather then to link to the proxy status object, this will you to delegate the sign / delete etc actions to the ca / status object instead see https://github.com/ohadlevy/foreman/commit/3422a9105199dd6f78a90f9a9a4cf77f4dcd1c10#diff-3bc418a60798c8ad4833276ef193c557